eldev.el

```diff
@@ -4028,6 +4031,221 @@ be passed to Emacs, else it will most likely fail."
+                                 container-eldev-cache-dir
+                                 eldev-global-cache-directory-name)))
+            (eldev--docker-local-dep-mounts container-home)
+            eldev-docker-run-extra-args
+            (cons img (funcall eldev--container-bootstrap-cmd-fn
+                               (mapconcat #'identity eldev-args " "))))))
+
+(defun eldev--docker-container-eldev-cmd (args)
+  "Return the eldev command to call in the docker container deduced from ARGS."
+  (car (eldev-filter (not (string-prefix-p "-" it)) args)))
+
+(defun eldev--docker-on-supported-os ()
+  "Return t if on a supported OS, else return nil."
+  (memq system-type '(gnu/linux gnu/kfreebsd darwin)))
+
+(eldev-defcommand eldev-docker (&rest parameters)
+  "Launch a specified Emacs version in a docker container.
+
+This command will execute the eldev command ELDEV_COMMAND against a
+specified Emacs version with the project loaded with all its
+dependencies in a docker container.
+
+GLOBAL_ARGS such as \"--trace\" may also precede ELDEV_COMMAND, in
+which case they will also be forwarded to the eldev call inside the
+container.
+
+VERSION must be a valid Emacs version, e.g. \"27.2\".
+
+A repository name/full image name may also be used in place of VERSION,
+in which case it will be fed directly to \"docker pull\" and
+\"docker run\".  Any string containing a \"/\" will be interpreted as
+such rather than an Emacs version.
+
+Note it will be assumed that Emacs and curl are installed on any image
+that is run.
+
+Command line arguments appearing after VERSION will be forwarded to an
+\"eldev\" call within the container.  For example:
+
+    eldev docker 25 emacs
+
+Will run \"eldev emacs\" inside an Emacs 25 container.
+
+The contents of `eldev-docker-run-extra-args' will be added to the
+\"docker run\" call this command makes.
+
+Emacs will not be started as a GUI unless the command is \"emacs\" and
+the \"--batch\" flag is not present.
+
+Note currently only linux and Mac OS systems are supported by this command."
+  :parameters     "VERSION [GLOBAL_ARGS..] ELDEV_COMMAND [ARGS...]"
+  :aliases        emacs-docker
+  :custom-parsing t
+  (unless (eldev--docker-on-supported-os)
+    (signal 'eldev-error
+            `(t ,(format eldev--docker-os-error-fmt-string system-type))))
+  (unless (car parameters)
+    (signal 'eldev-wrong-command-usage `(t "version not specified")))
+  (let* ((img (eldev--docker-determine-img (car parameters)))
+         (docker-exec (eldev-docker-executable))
+         (escaped-params (mapcar #'eldev-quote-sh-string (cdr parameters)))
+         (container-cmd (eldev--docker-container-eldev-cmd escaped-params))
+         (as-gui (and (string= "emacs" container-cmd)
+                      (not (member "--batch" parameters))))
+         (eldev--container-bootstrap-cmd-fn
+          (if (getenv "ELDEV_LOCAL")
+              (apply-partially
+               #'eldev--container-eldev-source-install-cmd "/eldev")
```

Comment:
Is this correct? I don't see the result of `(getenv "ELDEV_LOCAL")` being used anywhere. Maybe I just don't follow the logic.

Reply:
The function `eldev--conatainer-eldev-source-install-cmd` is only concerned with the location of the eldev source on the container and assumes that it's mounted in `/eldev`.  In this commit, I was using `(getenv "ELDEV_LOCAL")` as an indicator and calling it again when the mounts are determined, but this is lazy, the most recent changes have it so that `(getenv "ELDEV_LOCAL")` is called only once and its value passed through.